maplibre/src/render/resource/surface.rs

```diff
@@ -98,32 +98,32 @@ impl BufferedTextureHead {
         self.output_buffer.unmap();
     }
 
-    pub fn write_png<'a>(&self, padded_buffer: &wgpu::BufferView<'a>, png_output_path: &str) {
+    pub fn write_png<'a>(
+        &self,
+        padded_buffer: &wgpu::BufferView<'a>,
+        png_output_path: &str,
+    ) -> Result<(), crate::render::error::RenderError> {
```

Comment:
Ideally errors are kept very local. So at least for every `impl {}` block/struct, and maybe even for every function a new error type should be created.

I'm trying to follow the guidelines here :) https://mmapped.blog/posts/12-rust-error-handling.html#errors-problem-vs-solution